### PR TITLE
H222: Tau_x loss upweight 2.0 — target WSS_x basin disruption

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

**H222: Tau_x explicit channel upweighting — directly address WSS_x basin disruption**

Finding Q established that TTA provides no slope recovery for WSS_x (the only channel). The basin disruption is a training-trajectory property. If the training loss underweights WSS_x vs WSS_y/WSS_z, the optimizer naturally finds basins that sacrifice WSS_x precision.

**This is the only known mechanism that targets WSS_x at training time.** (TTA can't fix it; weight-arithmetic can't fix it.)

**Prediction**: Explicitly upweighting tau_x (2× vs default) should:
- Increase WSS_x gradient signal
- Pull optimizer toward basins with negative WSS_x slope (intact basin)
- Possibly trade small val_abupt regression for large WSS_x improvement

## Instructions

**Training-heavy sprint. ~3.5h training + 5min TTA eval. DDP=8 GPUs.**

### Step 1: Check for tau-x-loss-weight flag

```bash
python target/train.py --help 2>&1 | grep -iE "tau_x|wss_x|channel"
```

If `--tau-x-loss-weight` exists, use it. If not, add a 1-line patch to per-channel loss dict (probably `target/model/loss.py`). Do NOT touch protected files (data/loader.py, data/preload.py, data/split_manifest.json).

### Step 2: Full retrain with tau_x=2.0

```bash
SENPAI_TIMEOUT_MINUTES=240 torchrun --standalone --nproc-per-node=8 target/train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --manifest data/split_manifest.json \
  --agent edward \
  --optimizer lion --lion-beta1 0.9 --lion-beta2 0.99 \
  --lr 9e-5 --weight-decay 5e-4 --batch-size 4 \
  --tau-x-loss-weight 2.0 --tau-y-loss-weight 3.0 --tau-z-loss-weight 2.0 \
  --surface-loss-weight 2.0 --volume-loss-weight 0.5 \
  --use-surf-to-vol-xattn --enable-residual-positions \
  --epochs 13 --lr-warmup-epochs 1 --lr-schedule cosine \
  --ema-decay 0.999 --grad-clip 1.0 --save-best-checkpoint \
  --mirror-augment-p 0.25 \
  --wandb_group h222-edward-tau_x_2 \
  --run_name edward/h222-tau_x_2
```

Identical to H185 except `--tau-x-loss-weight 2.0` (vs default 1.0).

### Step 3: WSS_x slope tracking — KEY DIAGNOSTIC

Save EP6, EP9, EP13 EMA checkpoints. Run quick val+test (no TTA) at each:

| EP | val_WSS_x | test_WSS_x | slope sign | basin health | val_abupt |
|---|---|---|---|---|---|
| 6 | ? | ? | ? | ? | ? |
| 9 | ? | ? | ? | ? | ? |
| 13 | ? | ? | ? | ? | ? |

**Target**: WSS_x slope NEGATIVE at EP13 (vs H185's +0.0509 flipped).

### Step 4: TTA eval at best EMA

```bash
torchrun --standalone --nproc-per-node=8 target/eval_tta_h209.py \
  --checkpoint outputs/<run-dir>/checkpoint_best.pt \
  --output-dir outputs/h222_eval \
  --wandb-group h222-edward-tau_x_2 \
  --wandb-name "edward/h222-tau_x_2-tta" \
  --batch-size 2
```

### Step 5: Full report + SOTA gate

PRIMARY: val_abupt < 5.9755 AND test_abupt < 5.8221
Paper-floor: test_VP ≤ 3.421 AND test_SP ≤ 3.577 AND test_WSS ≤ 6.727

Include all metrics, WSS_x slope table, and delta_mirror_WSS (per Finding S).

## Baseline

Current SOTA (PR #1382 H209, H185+TTA): val_abupt 5.9755, test_abupt 5.8221, test_WSS 6.7214
H185 raw EP13: val_abupt 6.0172, test_abupt 5.8613, WSS_x slope +0.0509 (flipped)
W&B run: `bx3t1vdw` / source checkpoint: `yw2a5dyl` EP13 EMA

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT edward:

Before launching, I want to flag several discrepancies between the PR command and the actual H185 (`yw2a5dyl`) recipe — and several flags that don't exist on current `tay`. Need your confirmation before a 4h training run.

## 1. Flags that DON'T exist on current tay/train.py

- `--mirror-augment-p 0.25` — mirror augmentation has never been merged to `tay`. It exists on `askeladd/h148-mirror-augmentation` and `fern/h183-mirror-aug-tau-y-3p0-compound` branches but not here. H185 used `--mirror-augmentation` (a boolean, prob 0.5 implicit). For p=0.25 I'd need to port the H183 code AND add a probability parameter.
- `--enable-residual-positions` — not in train.py. Was this from a different branch?
- `--save-best-checkpoint` — train.py already saves best-val checkpoint automatically (line 1314).
- `--lr-schedule cosine` — flag doesn't exist; current train.py uses `--lr-cosine-t-max 13` for cosine.
- `--run_name`/`--wandb_group` — actual flags are `--wandb-name`/`--wandb-group` (will rename).

I added `--tau-x-loss-weight` (1-line patch, mirrors `--tau-y-loss-weight` plumbing) — that's now available.

## 2. Tau weights mismatch with H185 actual

PR says "Identical to H185 except `--tau-x-loss-weight 2.0`" but:

| Param | H185 yw2a5dyl actual | PR command | 
|---|---:|---:|
| tau_y_loss_weight | 1.3 | **3.0** |
| tau_z_loss_weight | 1.67 | **2.0** |

The 3.0/2.0 numbers don't match any H185 run I can find in W&B.

## 3. Other H185 recipe items missing from PR command

H185 also used (from yw2a5dyl config):
- `--grad-clip-norm 0.5` (PR has `--grad-clip 1.0` — wrong flag name AND wrong value)
- `--vol-points-schedule 0:16384:3:32768:6:49152:9:65536`
- `--use-qk-norm`
- `--rff-num-features 16 --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0"`
- `--pos-encoding-mode string_separable`
- `--no-compile-model`
- `--kill-thresholds "10864:val_primary/abupt_axis_mean_rel_l2_pct<33.0,32592:val_primary/abupt_axis_mean_rel_l2_pct<8.5,48897:val_primary/abupt_axis_mean_rel_l2_pct<7.0"`

## Two interpretations — please pick one

**Interpretation A** (strict "identical to H185 except tau_x"): port mirror_augmentation from H183 branch (with `--mirror-augment-p 0.25`), use H185's actual tau_y=1.3, tau_z=1.67, AND add all the H185 recipe items above. This is the cleanest test of "tau_x effect on a mirror-trained basin".

**Interpretation B** (run the exact PR command after fixing flag-name typos): drop mirror augmentation, use tau_y=3.0, tau_z=2.0, lr=9e-5, batch-size=4, surface-loss=2.0, volume-loss=0.5, use-surf-to-vol-xattn, cosine schedule, EMA, grad-clip-norm 1.0. This is a much simpler recipe (no mirror, no QK-norm, no RFF, no schedule) and will NOT reproduce H185 baseline — basin landscape will be entirely different.

Strong preference for **Interpretation A** since the hypothesis explicitly hinges on disrupting H185's mirror-trained basin. Interpretation B tests something else entirely.

Please confirm A or B and I'll launch immediately. Will also need explicit OK to add `--mirror-augment-p` flag if A.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #1391 because ADVISOR — CLOSING WITH APOLOGY. The Round 4 assignment had two systemic errors that I (advisor) own:

**Error 1: Recipe spec did not match actual yw2a5dyl config.** I cited the H185 recipe as `tau_y=3.0, tau_z=2.0, --mirror-augment-p 0.25, --grad-clip 1.0, --lr-schedule cosine, --enable-residual-positions, --save-best-checkpoint`. Verified from W&B config on yw2a5dyl, the ACTUAL recipe is `tau_y=1.3, tau_z=1.67, mirror_augmentation=True (boolean), --grad-clip-norm 0.5, --lr-cosine-t-max 13`. The other flags don't exist on tay. Student catches were correct.

**Error 2: Budget infeasibility.** yw2a5dyl actual `total_train_minutes=874.4` (~14.6h on 8 GPUs). The 6h SENPAI_TIMEOUT_MINUTES cap CANNOT fit a 13-epoch H185-recipe retrain. The Round 4 assignment was incompatible with the budget cap from the start.

**Mirror augmentation also not on tay:** verified via grep — `--mirror-augmentation` lives on `askeladd/h148-mirror-augmentation` and `fern/h183-mirror-aug-tau-y-3p0-compound` branches, never merged. Reproducing H185 recipe requires cherry-pick PR first.

BASELINE.md has been corrected on tay (commit pending).

**Reassignment incoming**: H225 — TTA rot_x angle sweep θ ∈ {1°,2°,4°,8°} on H185 (eval-only, ~60min)

Thank you for catching the recipe discrepancies before launching. Apologies for the wasted setup time..
